lib/Driver/ToolChains.cpp

```diff
@@ -1689,6 +1708,7 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
     Arguments.push_back(context.Args.MakeArgString(StaticRuntimeLibPath));
 
     SmallString<128> linkFilePath = StaticRuntimeLibPath;
+    llvm::sys::path::remove_filename(linkFilePath); // remove arch name
     llvm::sys::path::append(linkFilePath, "static-executable-args.lnk");
```

Comment:
Are these not arch-specific?

Reply:
They currently are not. I could see them possibly being arch specific but right now we use the same libraries everywhere. 

static-executable and static-library are a little busted though and do some squirrely stuff. Some of the libs end up getting doubled in the linker script and with static-executable it ends up linking in all symbols and not allowing the linker to strip at all which ends up making huge binaries unnecessarily. Was going to poke this in a later commit because it doesn't seem to being doing anything too useful for Linux right now and it's something we will probably want on Fuchsia.